core/src/main/scala/com/acervera/osm4scala/DenseNodesIterator.scala

```diff
@@ -26,78 +26,115 @@
 package com.acervera.osm4scala
 
 import com.acervera.osm4scala.model.NodeEntity
-import org.openstreetmap.osmosis.osmbinary.osmformat.{DenseNodes, StringTable}
-
-object DenseNodesIterator {
-
-  def apply(osmosisStringTable: StringTable, osmosisDenseNode: DenseNodes): DenseNodesIterator =
-    new DenseNodesIterator(osmosisStringTable, osmosisDenseNode)
-
-}
+import com.acervera.osm4scala.utilities.DecompressUtils
+import com.acervera.osm4scala.utilities.DecompressUtils.{decompressChangeset, decompressCoord, decompressTimestamp, decompressUid, decompressUserSid, iteratorCheck}
+import org.openstreetmap.osmosis.osmbinary.osmformat.{DenseInfo, DenseNodes, StringTable}
 
 /**
   * Iterator over a DenseDataNode block.
-  * By default, lanOffset, longOffset and graularity is 0, 0 and 100 because I did not found pbf files with other values.
   *
   * @param osmosisStringTable
   * @param osmosisDenseNode
   * @param latOffset
   * @param lonOffset
   * @param granularity
+  * @param dateGranularity
   */
-class DenseNodesIterator(osmosisStringTable: StringTable,
+case class DenseNodesIterator(osmosisStringTable: StringTable,
                          osmosisDenseNode: DenseNodes,
-                         latOffset: Long = 0,
-                         lonOffset: Long = 0,
-                         granularity: Int = 100)
+                         latOffset: Option[Long] = None,
+                         lonOffset: Option[Long] = None,
+                         granularity: Option[Int] = None,
+                         dateGranularity: Option[Int] = None)
     extends Iterator[NodeEntity] {
 
-  if (osmosisDenseNode.denseinfo.isDefined && osmosisDenseNode.denseinfo.get.visible.nonEmpty) {
-    throw new Exception("Only visible nodes are implemented.")
-  }
-
-  private val idIterator = osmosisDenseNode.id.iterator
-  private val lonIterator = osmosisDenseNode.lon.iterator
-  private val latIterator = osmosisDenseNode.lat.iterator
-  private val tagsIterator = osmosisDenseNode.keysVals.iterator
-
-  private var lastNode: NodeEntity = NodeEntity(0, 0, 0, Map())
+  private val idIterator: Iterator[Long] = osmosisDenseNode.id.iterator
```

Comment:
Let's allow Scala to infer types. The recommendation is to specify types that SCala can infer only for public functions and properties to help the API user.

Reply:
I still don't think letting compiler inferring all these is a good practice and it's safer and more consistent if we specify all these. But again, it's a personal interest. If you don't like it I can remove it.

Reply:
At least this is what I do when I work. To avoid bad Any happens and clearer to other readers